pkg/resources/file_format.go

```diff
@@ -0,0 +1,1128 @@
+package resources
+
+import (
+	"database/sql"
+	"fmt"
+
+	"github.com/chanzuckerberg/terraform-provider-snowflake/pkg/snowflake"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/pkg/errors"
+)
+
+// valid format type options for each File Format Type
+// https://docs.snowflake.com/en/sql-reference/sql/create-file-format.html#syntax
+var formatTypeOptions = map[string][]string{
+	"CSV": {
+		"compression",
+		"record_delimiter",
+		"field_delimiter",
+		"file_extension",
+		"skip_header",
+		"skip_blank_lines",
+		"date_format",
+		"time_format",
+		"timestamp_format",
+		"binary_format",
+		"escape",
+		"escape_unenclosed_field",
+		"trim_space",
+		"field_optionally_enclosed_by",
+		"null_if",
+		"error_on_column_count_mismatch",
+		"replace_invalid_characters",
+		"validate_utf8",
+		"empty_field_as_null",
+		"skip_byte_order_mark",
+		"encoding",
+	},
+	"JSON": {
+		"compression",
+		"file_extension",
+		"date_format",
+		"time_format",
+		"timestamp_format",
+		"binary_format",
+		"trim_space",
+		"null_if",
+		"enable_octal",
+		"allow_duplicate",
+		"strip_outer_array",
+		"strip_null_values",
+		"ignore_utf8_errors",
+		"replace_invalid_characters",
+		"skip_byte_order_mark",
+	},
+	"AVRO": {
+		"compression",
+		"trim_space",
+		"null_if",
+	},
+	"ORC": {
+		"trim_space",
+		"null_if",
+	},
+	"PARQUET": {
+		"compression",
+		"snappy_compression",
+		"binary_as_text",
+		"trim_space",
+		"null_if",
+	},
+	"XML": {
+		"compression",
+		"ignore_utf8_errors",
+		"preserve_space",
+		"strip_outer_element",
+		"disable_snowflake_data",
+		"disable_auto_convert",
+		"skip_byte_order_mark",
+		"trim_space",
+		"null_if",
```

Comment:
I don't think there is `trim_space` or `null_if` for XML... looking at this list: https://docs.snowflake.com/en/sql-reference/sql/create-file-format.html#syntax